src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -484,6 +508,61 @@ async def _genomic_to_transcript_exon_coordinate(
 
         params = {key: None for key in TranscriptExonData.model_fields}
 
+        if get_nearest_transcript_junction:
+            # Check if is_fusion_transcript_segment is given
+            alt_acs, w = self.seqrepo_access.chromosome_to_acs(chromosome)
+            if not alt_acs:
+                return self._return_warnings(resp, w)
+            alt_ac = alt_acs[0]
+            mane_transcripts = self.mane_transcript_mappings.get_gene_mane_data(gene)
+            if mane_transcripts:
+                transcript = mane_transcripts[0]["RefSeq_nuc"]
+            else:
+                # Attempt to find a coding transcript if a MANE transcript cannot be found
+                results = await self.uta_db.get_transcripts(gene=gene, alt_ac=alt_ac)
+                if not results.is_empty():
+                    transcript = results[0]["tx_ac"][0]
+                else:
+                    # Run if gene is for a noncoding transcript
+                    query = f"""
+                        SELECT DISTINCT tx_ac
+                        FROM {self.uta_db.schema}.tx_exon_aln_v
+                        WHERE hgnc = '{gene}'
+                        AND alt_ac = '{alt_ac}'
+                        """  # noqa: S608
+                    result = await self.uta_db.execute_query(query)
+                    if result:
+                        transcript = result[0]["tx_ac"]
+                    else:
+                        return self._return_warnings(
+                            resp, f"Could not find a transcript for {gene} on {alt_ac}"
+                        )
+
+            tx_genomic_coords = await self.uta_db.get_tx_exons_genomic_coords(
+                tx_ac=transcript, gene=gene, alt_ac=alt_ac, strand=strand
+            )
+            # Check if breakpoint occurs on an exon. If not, determine the adjacent exon given the selected transcript
+            if not self._is_exonic_breakpoint(pos, tx_genomic_coords[0]):
```

Comment:
Are there tests for both? If so, could you make a comment on the tests to state whether or not breakpoint occurs on an exon?